searx/utils.py

```diff
@@ -475,23 +471,23 @@ def ecma_unescape(s):
         'ó'
     """
     # "%u5409" becomes "吉"
-    s = ecma_unescape4_re.sub(lambda e: chr(int(e.group(1), 16)), s)
+    string = _ECMA_UNESCAPE4_RE.sub(lambda e: chr(int(e.group(1), 16)), string)
     # "%20" becomes " ", "%F3" becomes "ó"
-    s = ecma_unescape2_re.sub(lambda e: chr(int(e.group(1), 16)), s)
-    return s
+    string = _ECMA_UNESCAPE2_RE.sub(lambda e: chr(int(e.group(1), 16)), string)
+    return string
 
 
-def get_string_replaces_function(replaces):
+def get_string_replaces_function(replaces: Dict[str, str]) -> Callable:
```

Comment:
I don't understand why you keep specifying types that are less accurate than what Pyright can infer without any annotations. Such annotations do more harm than good.

Pyright infers the return type to be `((text: Unknown) -> str)`.

Reply:
Fixed (?)
Feel free to add other commits in this PR to fix the typing issues.